daemon/containerd/image_list.go

```diff
@@ -91,12 +93,40 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 			return nil, err
 		}
 
+		var repoTags, repoDigests []string
+		rawImg := img.Metadata()
+		target := rawImg.Target.Digest
+
+		logger := logrus.WithFields(logrus.Fields{
+			"name":   img.Name(),
+			"digest": target,
+		})
+
+		ref, err := reference.ParseNamed(rawImg.Name)
+		if err == nil {
+			repoTags = append(repoTags, reference.TagNameOnly(ref).String())
+
+			digested, err := reference.WithDigest(reference.TrimNamed(ref), target)
+			if err != nil {
+				logger.WithError(err).Error("failed to create digested reference")
+			} else {
+				repoDigests = append(repoDigests, digested.String())
+			}
+		}
+
+		if err != nil {
+			if !isDanglingImage(rawImg) {
+				logger.WithError(err).Error("failed to parse image name as reference")
```

Comment:
This log looks out of place?

Reply:
It's there to issue an error in case we encounter an image like "🐳😄🐱", see: https://github.com/moby/moby/pull/44840#discussion_r1084157945
We don't want this error to be logged for dangling images - which have `moby-dangling@sha256:...` format, so they're not a valid Named reference (missing registry).